### ISSUE TITLE
Node relays transaction broadcast, but does not store tx

### ISSUE THREAD

**Issue by kantai** [blockstack/stacks-blockchain#1941]
A transaction in the testnet got "stuck" in the mempool, because while the node that received the transaction broadcasted it to a peer, and that peer relayed the message, the miner never appeared to store the transaction or consider it for inclusion in a block.

The tx in question here is `0xb039b12c545aa7affbddd525e5690c25a0210ddf1e381be2b6495671f9d599c3`, two sets of logs are attached: first is all the logs containing that txid, and second is the surrounding few seconds of logs on the miner node.

[relay-not-store-nearby-logs.txt](https://github.com/blockstack/stacks-blockchain/files/5319049/relay-not-store-nearby-logs.txt)
[relay-not-store.txt](https://github.com/blockstack/stacks-blockchain/files/5319051/relay-not-store.txt)

**Comment by jcnelson** [comment]
In `relay-not-store.txt`, I see:

```
"Oct 1, 2020 @ 00:01:16.615","stacks-node-follower","DEBUG [1601510476.614] [src/net/relay.rs:876] [ThreadId(6)] Stored tx b039b12c545aa7affbddd525e5690c25a0210ddf1e381be2b6495671f9d599c3
```

If you look at the code around `relay.rs:876`, you'll see:

```
    /// Store a single transaction
    /// Return true if stored; false if it was a dup
    fn store_transaction(
        mempool: &mut MemPoolDB,
        consensus_hash: &ConsensusHash,
        block_hash: &BlockHeaderHash,
        tx: StacksTransaction,
    ) -> bool {
        let txid = tx.txid();
        if mempool.has_tx(&txid) {
            return true;
        }

        if let Err(e) = mempool.submit(consensus_hash, block_hash, tx) {
            info!("Reject transaction {}: {:?}", txid, &e);
            return false;
        }

        debug!("Stored tx {}", txid);
        return true;
    }
```

If you _don't_ see anything about storing the TX, it may be because it's already in the mempool.  Can you inspect the mempool DB and confirm?

**Comment by psq** [comment]
@katnai 0xb039b12c545aa7affbddd525e5690c25a0210ddf1e381be2b6495671f9d599c3 was my transaction, calling `stack-stx, which was a second transaction I sent quickly after a previous one (with the correct nonce), the first transaction from the same address was trying to stack an amount larger than the balance, and was going to get rejected at execution. Not sure that matters, but could have been that should have been included in the same block

**Comment by diwakergupta** [comment]
Fixed with #1980
